### PR TITLE
Chat autocomplete

### PR DESCRIPTION
**Pull request by palmfjord**

### CODE REVIEW

File: src/main/java/com/hedvig/backoffice/services/hopeAutocomplete/HopeAutocompleteServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.hedvig.backoffice.services.hopeAutocomplete;
+
+import com.hedvig.backoffice.services.hopeAutocomplete.dto.AutocompleteSuggestion;
+import com.hedvig.backoffice.services.hopeAutocomplete.dto.RemoteAutocompleteSuggestionRequestBodyDto;
+import com.hedvig.backoffice.services.hopeAutocomplete.dto.SelectSuggestionRequestBodyDto;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/services/hopeAutocomplete/HopeAutocompleteServiceImpl.java]: Shall we use the `@Service ` annotation here?

File: src/main/java/com/hedvig/backoffice/web/AutocompleteController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  public AutocompleteController(final HopeAutocompleteService autocompleteService) {
+    this.autocompleteService = requireNonNull(autocompleteService);
+  }
+
+  @GetMapping(path = {"/suggestions"})
+  public List<AutocompleteSuggestion> getSuggestions(@RequestParam("query") final String query) {
+    return autocompleteService.getAutocompleteSuggestions(query);
+  }
+
+  @PostMapping(path = {"select"})
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/web/AutocompleteController.java]: i guess a `/` is needed before `select` :grin:

File: src/main/java/com/hedvig/backoffice/services/hopeAutocomplete/HopeAutocompleteServiceImpl.java (hunk 1 above)
Comment (palmfjord) [reply src/main/java/com/hedvig/backoffice/services/hopeAutocomplete/HopeAutocompleteServiceImpl.java]: idk, it's instantiated manually [here](https://github.com/HedvigInsurance/back-office/pull/175/files/ff4cd7f637fc884ca1e42d8bc8585fa74509f866#diff-e2794d21aa5f65a6d09bc285ec528e1eR58) so maybe its not needed?

File: src/main/java/com/hedvig/backoffice/web/AutocompleteController.java (hunk 2 above)
Comment (palmfjord) [reply src/main/java/com/hedvig/backoffice/web/AutocompleteController.java]: nice catch!

**Pull request merged**